core/job/service/job_service.go

```diff
@@ -983,9 +1066,10 @@ func (j *JobService) generateJobs(ctx context.Context, tenantWithDetails *tenant
 			return func() (interface{}, error) {
 				generatedJob, err := j.generateJob(ctx, tenantWithDetails, currentSpec)
 				if err != nil {
-					j.logger.Error("error generating job [%s]: %s", currentSpec.Name(), err)
+					errorMsg := fmt.Sprintf("error generating job [%s]: %s", spec.Name().String(), err.Error())
```

Comment:
is it expected to change `currentSpec` to `spec`? in this case `spec` is outside the function, doesn't it lead to unpredicted behavior when running concurrently?

Reply:
yes, it should be `currentSpec`. thank you for noticing this.

Reply:
updated